projects/ad7616-st/src/app.c

```diff
@@ -0,0 +1,134 @@
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+#include <stdio.h>
+
+#include "app.h"
+#include "parameters.h"
+#include "iio.h"
+#include "ad7616.h"
+#include "iio_ad7616.h"
+#include "no_os_util.h"
+
+extern struct no_os_uart_desc *uart_desc;
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+/***************************************************************************//**
+ * @brief Basic example main execution.
+ *
+ * @return ret - Result of the example execution. If working correctly, will
+ *               execute continuously the while(1) loop and will not return.
+*******************************************************************************/
+int iio_ad7616_initialize(struct ad7616_dev *dev, struct iio_desc **iio_desc)
+{
+	int ret;
+	struct iio_device *iio_dev;
+
+	ad7616_iio_param_init(&iio_dev);
+
+	struct iio_device_init iio_device_init = { .name = "ad7616", .dev = dev, .dev_descriptor = iio_dev };
+
+	struct ad7616_sequencer_layer layers[] = {
+		{ AD7616_VA0, AD7616_VB0 },
+		{ AD7616_VA1, AD7616_VB1 },
+		{ AD7616_VA2, AD7616_VB2 },
+		{ AD7616_VA3, AD7616_VB3 },
+		{ AD7616_VA4, AD7616_VB4 },
+		{ AD7616_VA5, AD7616_VB5 },
+		{ AD7616_VA6, AD7616_VB6 },
+		{ AD7616_VA7, AD7616_VB7 },
+	};
+
+	/* IIO interface init parameters */
+	static struct iio_init_param iio_init_params = {
+		.phy_type = USE_UART,
+		.nb_devs = 1,
+		// .uart_desc = uart_desc, // Not a constant...
+		// .devs = &iio_device_init
+	};
+	iio_init_params.uart_desc = uart_desc;
+	iio_init_params.devs = &iio_device_init;
+
+	ad7616_setup_sequencer(dev, layers, NO_OS_ARRAY_SIZE(layers), 1);
+
+	struct ad7616_conversion_result results[8] = {0};
+	ret = ad7616_read_data_serial(dev, results, 1);
+	if (ret)
+		return ret;
+
+	printf("Sequencer + burst\r\n");
+	for (uint32_t i = 0; i < 8; i++) {
+		printf("Channel A[%lu] = %x\r\n", i, results[i].channel_a);
+		printf("Channel B[%lu] = %x\r\n", i, results[i].channel_b);
+	}
+
+	return iio_init(iio_desc, &iio_init_params);
+}
+
+int iio_firmware_main()
+{
+	struct ad7616_dev *ad7616_dev;
+	struct iio_desc *iio_desc;
+	int ret;
+
+	// ad7616_ip.crc = 1; // If you want to use CRC, uncomment this
+
+	ret = ad7616_setup(&ad7616_dev, &ad7616_ip);
+	if (ret) {
+		printf("AD7616 failed to instantiate\r\n");
```

Comment:
use `https://github.com/analogdevicesinc/no-OS/blob/master/include/no_os_print_log.h` -> `pr_err` / `pr_info` depending on the case.